siriuspy/siriuspy/cycle/conn.py

```diff
@@ -753,3 +754,148 @@ def _get_duration_and_waveform(self):
+        'PSAmpOverCurrFlagR-Sts', 'PSAmpOverTempFlagR-Sts',
+        'TestOpenLoopTriang-Sel', 'TestOpenLoopTriang-Sts',
+        'TestOpenLoopSquare-Sel', 'TestOpenLoopSquare-Sts',
+        'TestClosedLoopSquare-Sel', 'TestClosedLoopSquare-Sts',
+    ]
+
+    def __init__(self, psname):
+        """Constructor."""
+        self._psname = _PVName(psname)
+        self._waveform = None
+        self._cycle_duration = None
+        self._times = None
+        self._pvs = dict()
+        for prop in FOFBPSCycler.properties:
+            if prop not in self._pvs.keys():
+                self._pvs[prop] = _PV(
+                    self._psname.substitute(prefix=VACA_PREFIX, propty=prop),
+                    connection_timeout=TIMEOUT_CONNECTION)
+
+    @property
+    def psname(self):
+        """Power supply name."""
+        return self._psname
+
+    @property
+    def connected(self):
+        """Return connected state."""
+        for prop in FOFBPSCycler.properties:
+            if not self[prop].connected:
+                return False
+        return True
+
+    def wait_for_connection(self, timeout=0.5):
+        """Wait for connection."""
+        for pvobj in self._pvs.values():
+            if not pvobj.wait_for_connection(timeout):
+                return False
+        return True
+
+    @property
+    def waveform(self):
+        """Return waveform."""
+        if self._waveform is None:
+            self._get_duration_and_waveform()
+        return self._waveform
+
+    def cycle_duration(self, _):
+        """Return the duration of the cycling in seconds."""
+        if self._cycle_duration is None:
+            self._get_duration_and_waveform()
+        return self._cycle_duration
+
+    def check_intlks(self, wait=2):
+        """Check interlocks."""
+        if not self.connected:
+            return False
+        status = (self._pvs['PSAmpOverCurrFlagL-Sts'].value == 1)
+        status &= (self._pvs['PSAmpOverTempFlagL-Sts'].value == 1)
+        status &= (self._pvs['PSAmpOverCurrFlagR-Sts'].value == 1)
+        status &= (self._pvs['PSAmpOverTempFlagR-Sts'].value == 1)
+        return status
+
+    def check_on(self):
+        """Return whether power supply PS is on."""
+        return _pv_timed_get(self['PwrState-Sts'], 1)
+
+    def set_current_zero(self):
+        """Set PS current to zero ."""
+        return _pv_conn_put(self['Current-SP'], 0)
+
+    def check_current_zero(self, wait=5):
+        """Return whether power supply PS current is zero."""
+        return _pv_timed_get(self['Current-RB'], 0, abs_tol=0.01, wait=wait)
+
+    def prepare(self, _):
+        """Config power supply to cycling mode."""
+        status = True
+        if not self.check_current_zero(wait=0.5):
+            status &= self.set_current_zero()
+        return status
+
+    def is_prepared(self, _, wait=5):
+        """Return whether power supply is ready."""
+        status = self.check_current_zero(wait)
+        return status
```

Comment:
in this case you can do `return self.check_current_zero(wait)` without the need of `status` variable, right?

Reply:
right, done!